config/config_test.go

```diff
@@ -1,34 +1,295 @@
+type YamlConfigTest struct {
+	YamlConfig
+}
+
+// NewYamlConfigTest tests func for updating fields values in tests cases
+func (t *YamlConfigTest) NewYamlConfigTest() *YamlConfig {
+	t.YamlConfig = prepareYamlConfig(
+		"100MB", 31, 45, "127.0.0.1:81", ":80", "client1", []string{"dev"})
+	return &t.YamlConfig
+}
+
+func TestYAMLUrlParsingSuccessful(t *testing.T) {
 	correct := []byte(`field: http://golang.org:80/pkg/net`)
 	testyaml := TestYaml{}
 	err := yaml.Unmarshal(correct, &testyaml)
 	assert.NoError(t, err, "Should be correct")
 }
 
-func TestYAMLURLParsingFailure(t *testing.T) {
+func TestYAMLUrlParsingFailure(t *testing.T) {
 	incorrect := []byte(`field: golang.org:80/pkg/net`)
 	testyaml := TestYaml{}
 	err := yaml.Unmarshal(incorrect, &testyaml)
 	assert.Error(t, err, "Missing protocol should return error")
 }
 
-func TestYAMLURLParsingEmpty(t *testing.T) {
-	incorrect := []byte(`field:`)
+func TestYAMLUrlParsingEmpty(t *testing.T) {
+	incorrect := []byte(`field: "1"`)
+	testyaml := shardingconfig.YAMLUrl{}
+	err := yaml.Unmarshal(incorrect, &testyaml)
+	assert.Error(t, err, "Should not even try to parse")
+	assert.Nil(t, testyaml.URL, "Should be nil")
+}
+
+func TestListenYamlParameterValidation(t *testing.T) {
+	incorrect := []byte(`Listen: ":8080"`)
 	testyaml := TestYaml{}
 	err := yaml.Unmarshal(incorrect, &testyaml)
 	assert.NoError(t, err, "Should not even try to parse")
 	assert.Nil(t, testyaml.Field.URL, "Should be nil")
 }
+
+func TestShouldValidateListenConf(t *testing.T) {
+	var testConf YamlConfigTest
+	testListenData := []string{"127.0.0.1:8080", ":8080", ":80"}
+
+	for _, listenValue := range testListenData {
+		testConf.NewYamlConfigTest().Listen = listenValue
+		result, _ := ValidateConf(testConf.YamlConfig)
+		assert.True(t, result, "Should be true")
+	}
+}
+func TestShouldNotValidateListenConf(t *testing.T) {
+	var testConf YamlConfigTest
+	testWrongListenData := []string{"", "-", " ", "aaa", ":bbb", "c:"}
+
+	for _, listenWrongValue := range testWrongListenData {
+		testConf.NewYamlConfigTest().Listen = listenWrongValue
+		result, _ := ValidateConf(testConf.YamlConfig)
+		assert.False(t, result, "Should be false")
+	}
+}
+
+func TestShouldValidateConfWithRegexp(t *testing.T) {
+	var testConf YamlConfigTest
+	testConf.NewYamlConfigTest().BodyMaxSize = "40MB"
+
+	result, _ := ValidateConf(testConf.YamlConfig)
+
+	assert.True(t, result, "Should be true")
+}
+
+func TestShouldNotValidateConfWithWrongBodyMaxSizeValue(t *testing.T) {
+	var testConf YamlConfigTest
+	testConf.NewYamlConfigTest().BodyMaxSize = "0"
+
+	result, validationErrors := ValidateConf(testConf.YamlConfig)
+
+	assert.Contains(t, validationErrors, "BodyMaxSize")
+	assert.False(t, result, "Should be false")
+}
+
+func TestShouldValidateConfMaintainedBackendWhenNotEmpty(t *testing.T) {
+	maintainedBackendHost := "127.0.0.1:85"
+	testConfData := prepareYamlConfig("112MB", 21, 32, maintainedBackendHost, ":80", "client1", []string{"dev"})
```

Comment:
because it's one line of code in this case (var maintainedBackendHost name it's deep field in structure and preparing this structure need 4+ lines of code) e.g.:

```
	var testConf YamlConfigTest
	maintainedBackendHost := "127.0.0.1:85"
	url2 := url.URL{Scheme: "http", Host: maintainedBackendHost}
	maintainedBackends := []shardingconfig.YAMLUrl{{&url2}}
	testConf.NewYamlConfigTest().MaintainedBackends = maintainedBackends
```